examples/DynamicCantileverCase/dynamic_cantilever.py

```diff
@@ -0,0 +1,145 @@
+        base_length,
+        base_area,
+        moment_of_inertia,
+        youngs_modulus,
+        density,
+        mode=mode,
+        end_velocity=end_velocity,
+    )
+
+    initial_velocity = vibration.get_initial_velocity_profile(
+        cantilever.position_collection[0, :]
+    )
+    cantilever.velocity_collection[2, :] = initial_velocity
+
+    # Add call backs
+    class CantileverCallBack(CallBackBaseClass):
+        def __init__(self, step_skip: int, callback_params: dict):
+            CallBackBaseClass.__init__(self)
+            self.every = step_skip
+            self.callback_params = callback_params
+
+        def make_callback(self, system, time, current_step: int):
+
+            if current_step % self.every == 0:
+
+                self.callback_params["time"].append(time)
+                self.callback_params["position"].append(
+                    system.position_collection.copy()
+                )
+                self.callback_params["deflection"].append(
+                    system.position_collection[2, -1].copy()
+                )
+                return
+
+    recorded_history = defaultdict(list)
+    cantilever_sim.collect_diagnostics(cantilever).using(
+        CantileverCallBack, step_skip=step_skips, callback_params=recorded_history
+    )
+    cantilever_sim.finalize()
+
+    total_steps = int(final_time / dt)
+    print(f"Total steps: {total_steps}")
+
+    timestepper = PositionVerlet()
+
+    integrate(
+        timestepper,
+        cantilever_sim,
+        final_time,
+        total_steps,
+    )
+
+    # FFT
+    amplitudes = np.abs(fft(recorded_history["deflection"]))
+    fft_length = len(amplitudes)
+    amplitudes = amplitudes * 2 / fft_length
+    omegas = fftfreq(fft_length, dt * step_skips) * 2 * np.pi  # [rad/s]
+
+    try:
+        peaks, _ = find_peaks(amplitudes)
+        peak = peaks[np.argmax(amplitudes[peaks])]
+
+        simulated_frequency = omegas[peak]
+        theoretical_frequency = vibration.get_omega()
+
+        simulated_amplitude = max(recorded_history["deflection"])
+        theoretical_amplitude = vibration.get_amplitude()
+
+        print(
+            f"Theoretical frequency: {theoretical_frequency} rad/s \n"
+            f"Simulated frequency: {simulated_frequency} rad/s \n"
+            f"Theoretical amplitude: {theoretical_amplitude} m \n"
+            f"Simulated amplitude: {simulated_amplitude} m"
+        )
+
+        return {
+            "rod": cantilever,
+            "recorded_history": recorded_history,
+            "fft_frequencies": omegas,
+            "fft_amplitudes": amplitudes,
+            "vibration": vibration,
+            "peak": peak,
+            "simulated_frequency": simulated_frequency,
+            "theoretical_frequency": theoretical_frequency,
+            "simulated_amplitude": simulated_amplitude,
+            "theoretical_amplitude": theoretical_amplitude,
+        }
+
+    except IndexError:
```

Comment:
Is this error correct? What about we do `RuntimeError` ? @bhosale2 any thoughts?

Reply:
I agree `RuntimeError`maybe more apt.

Reply:
[826eeba](https://github.com/GazzolaLab/PyElastica/pull/173/commits/826eeba5c6226b53c79145f7fa247dcf3b1fedda)